packages/components/src/icon-button/index.js

```diff
@@ -20,7 +20,7 @@ import Dashicon from '../dashicon';
 // is common to apply a ref to the button element (only supported in class)
 class IconButton extends Component {
 	render() {
-		const { icon, children, label, className, tooltip, focus, shortcut, ...additionalProps } = this.props;
+		const { icon, children, label, className, tooltip, dotTip, focus, shortcut, ...additionalProps } = this.props;
```

Comment:
I don't like that the component button is aware of a "DotTip" special component. Can't we just pass `tooltip` to force showing the tooltips?

Reply:
Hi @youknowriad, I tried that before, but it caused problems. The problem is that if a button has tooltip and a dotTip we have an erratic behavior because both components catch the same events. DotTip and tooltip cannot be descendants of each other in the dom tree.
To solve that I added a verification that checks if both dotTip and tooltips exist to show the components as siblings.
I understand it is not the best scenario to have button aware of dotTips, but it given that button is mindful of tooltip it may make some sense also to have button aware of the dotTip given that both components are similar.

Reply:
🤔 this is a pickle. I'm trying to think of alternatives. Do you know why `IconButton` doesn't render its tooltip if there are children? I assume this was added so as to fix some bug—is there an alternative fix?

Otherwise, could we maybe just have the `IconButton` and `DotTip` be siblings within a dummy element?

```js
<span>
	<IconButton />
	<DotTip />
</span>
```

It's still gross but I think I prefer having a one-off `<span>` than to having all `IconButton`s have a encalsuplation-breaking `dotTip` property.